src/track/beats.cpp

```diff
@@ -449,6 +449,22 @@ Beats::ConstIterator Beats::iteratorFrom(audio::FramePos position) const {
     return it;
 }
 
+Beats::ConstIterator Beats::iteratorClosestTo(audio::FramePos position) const {
+    auto it = iteratorFrom(position);
+    if (it == cbegin()) {
+        return it;
+    }
+
+    const auto deltaFrames = *it - position;
+    it--;
+    if ((position - *it) < deltaFrames) {
+        return it;
+    }
+
+    it++;
```

Comment:
I think it make sense to cache the next iterator. Than we can give theme reasonable names to improve the readability as well.